File: generate_results_python_r/real_data_processing.py
"""
Script used to perform filtering of SNVs, tree inference and parameter optimization on cancer data
-> run generate_results_cpp/real_data_processing.cpp for faster results instead.
"""

from src_python.generate_results import generate_sciterna_results

import yaml

with open("../config/config.yaml", "r") as file:
    config = yaml.safe_load(file)

bootstrap_samples = 1000  # number of bootstrap samples if use_bootstrap=True
use_bootstrap = False  # use bootstrapping of mutations or not
n_snps = 300  # number of SNVs if method="first_k"
posterior_cutoff = 0.05  # posterior likelihood of locus being mutated if method="threshold"
method= "threshold"  # "first_k" "threshold" "highest_post"
n_rounds = 2  # Number of rounds of tree optimization and parameter estimation
sample = "mm34"  # sample name
flipped_mutation_direction = True  # if true allow the root genotype to be learned during tree optimization
only_preprocessing = False  # if true only run mutation filtering, no tree inference
tree_space = ["c", "m"]  # "c" = cell tree space, "m" = mutation tree space
reshuffle_nodes = False  # Prune and reinsert individual nodes in the mutation tree
ref_to_alt = False  # if True allow SNVs to only mutate ref -> alt

input_path = rf"../data/input_data/{sample}"
output_path = rf"../data/results/{sample}/sciterna"

generate_sciterna_results(path=input_path, pathout=output_path,
                          n_bootstrap=bootstrap_samples, use_bootstrap=use_bootstrap, tree_space=tree_space,
                          flipped_mutation_direction=flipped_mutation_direction, n_keep=n_snps, posterior_cutoff=posterior_cutoff,
                          n_rounds=n_rounds, only_preprocessing=only_preprocessing, method=method, reshuffle_nodes=reshuffle_nodes,
                          ref_to_alt=ref_to_alt)

File: generate_results_cpp/real_data_processing.cpp
/*
Script used to run SCITE-RNA on real cancer datasets.
*/

#include <vector>
#include <string>
#include <mutation_filter.h>

#include "utils.h"
#include <mutation_filter.h>
#include "generate_results.h"


int main() {
    int bootstrap_samples = 1000; // how many bootstrap samples to use for tree inference, only used if use_bootstrap is true
    bool use_bootstrap = true; // if true, bootstrap samples of the SNVs are used, otherwise the results are computed from the original data
    int n_snps = 3000; // how many mutations to select for tree inference, only used if method is "first_k"
    double posterior_threshold = 0.05; // threshold for filtering mutations, only used if method is "threshold"
    std::string method = "threshold"; // criterion for filtering mutations, can be "threshold" (loci with posterior probability of being mutated > threshold are selected) , "first_k" (k loci with highest posterior probability of being mutated are selected) or "highest_post" (loci where highest probability is a mutation and not no mutation are selected)
    int n_rounds = 2; // how many rounds of tree inference and parameter optimization to perform
    std::string sample = "mm16"; // specify the sample name, e.g. "mm16", "mm34"
    bool flipped_mutation_direction = true; // if true, we allow the model to switch the root genotype/mutation direction during tree inference
    bool only_preprocessing = false; // if true, only the mutation filtering step is performed, no tree inference or parameter optimization
    std::vector<std::string> tree_space = {"c", "m"}; // which tree spaces to use during optimization, "c" for cell lineage tree, "m" for mutation tree, the order determines in which space the optimization starts
    bool reshuffle_nodes = false; // false makes optimization faster for large numbers of mutations as individual nodes are not pruned and reinserted, instead only pruning and reattaching of subtrees is performed
    bool load_from_file = false; // if true, the selected mutations and genotypes are loaded from file, otherwise they are computed from the data
    bool reduced_output = false; // if true the number of output files is reduced, i.e. the genotype and mutation indicator files are not saved

//    std::string input_path = "../data/input_data/" + sample;
    std::string input_path = "/cluster/work/bewi/members/znorio/SCITE-RNA-v2/data/input_data/" + sample;
//    std::string output_path = "../data/results/" + sample + "/sciterna";
    std::string output_path = "/cluster/work/bewi/members/znorio/SCITE-RNA-v2/data/results/" + sample + "/sciterna";

    std::vector<std::vector<int>> ref = read_csv(input_path + "/ref.csv");
    std::vector<std::vector<int>> alt = read_csv(input_path + "/alt.csv");

    generate_sciterna_results(ref, alt, input_path, output_path,
                            bootstrap_samples, use_bootstrap, tree_space,
                            flipped_mutation_direction, n_snps, posterior_threshold,
                            n_rounds, only_preprocessing, method, reshuffle_nodes, load_from_file, reduced_output);
    return 0;
}
